### ISSUE TITLE
"Failed decryption event!"

### ISSUE THREAD

**Issue by mindbeat** [matrixgpt/matrix-chatgpt-bot#73]
I'm trying the bot on my server, but run into "Failed decryption event!"
15:34
Jan 29 15:14:27 matrix matrix-bot-chatgpt[80609]: Sun, 29 Jan 2023 14:14:27 GMT [INFO] [index] Starting bot...
Jan 29 15:14:27 matrix matrix-bot-chatgpt[80609]: Sun, 29 Jan 2023 14:14:27 GMT [INFO] [MatrixClientLite] End-to-end encryption enabled
Jan 29 15:14:27 matrix matrix-bot-chatgpt[80609]: Sun, 29 Jan 2023 14:14:27 GMT [INFO] [index] Bot started!
Jan 29 15:14:35 matrix matrix-bot-chatgpt[80609]: Sun, 29 Jan 2023 14:14:35 GMT [INFO] [index] Bot joined room !tIetnGTNGSVgidCSri:dnet.one
Jan 29 15:14:35 matrix matrix-bot-chatgpt[80609]: Sun, 29 Jan 2023 14:14:35 GMT [ERROR] [MatrixHttpClient] (REQ-10) { errcode: 'M_NOT_FOUND', error: 'Event not found.' }
Jan 29 15:14:35 matrix matrix-bot-chatgpt[80609]: Sun, 29 Jan 2023 14:14:35 GMT [ERROR] [MatrixHttpClient] (REQ-12) { errcode: 'M_NOT_FOUND', error: 'Event not found.' }
Jan 29 15:14:35 matrix matrix-bot-chatgpt[80609]: Sun, 29 Jan 2023 14:14:35 GMT [ERROR] [MatrixHttpClient] (REQ-11) { errcode: 'M_NOT_FOUND', error: 'Event not found.' }
Jan 29 15:14:35 matrix matrix-bot-chatgpt[80609]: Sun, 29 Jan 2023 14:14:35 GMT [ERROR] [MatrixHttpClient] (REQ-14) { errcode: 'M_NOT_FOUND', error: 'Event not found.' }
Jan 29 15:14:51 matrix matrix-bot-chatgpt[80609]: Sun, 29 Jan 2023 14:14:51 GMT [INFO] [index] Bot joined room !HNYBanlMJmWorvtfQS:dnet.one
Jan 29 15:14:52 matrix matrix-bot-chatgpt[80609]: Sun, 29 Jan 2023 14:14:52 GMT [ERROR] [MatrixClientLite] Decryption error on !HNYBanlMJmWorvtfQS:dnet.one $R5WSiS2xnsXTI1usZnYiWj9yl_PKHxDk7aJCQXXyJGE [Error: decryption failed because the room key is missing] {
Jan 29 15:14:52 matrix matrix-bot-chatgpt[80609]:   code: 'GenericFailure'
Jan 29 15:14:52 matrix matrix-bot-chatgpt[80609]: }
Jan 29 15:14:52 matrix matrix-bot-chatgpt[80609]: Sun, 29 Jan 2023 14:14:52 GMT [ERROR] [index] Failed decryption event!

Using ansible (https://github.com/spantaleev/matrix-docker-ansible-deploy)

**Comment by mindbeat** [comment]
I logged in with element web to get the matrix access token. This seems to have caused the problem. 

I got it fixed by following steps:

1. Clean up chatgpt mount folder.
$ rm /matrix/chatgpt/* -r

2. Remove docker image, run "docker images" to find ID
$ docker image rm [Image ID]

3. Delete bot user with synapse admin. 

4. Create new user
$ ansible-playbook -i inventory/hosts setup.yml --extra-vars='username=chatgpt password=PASSWORD admin=no' --tags=register-user --ask-become-pass

5. Add password to vars.yml, and #comment the old accesstoken. 
matrix_bot_chatgpt_matrix_bot_username_localpart: 'chatgpt'
matrix_bot_chatgpt_matrix_bot_password: 'PASSWORD'

6. Run ansible setup
$ ansible-playbook -i inventory/hosts setup.yml --tags=setup-all,start --ask-become-pass

7. Get the accesstoken from the log instead of login in through element. 
$ sudo journalctl -fu matrix-bot-chatgpt

8. Add accesstoken and remove password to vars.yml
matrix_bot_chatgpt_matrix_access_token:'ACCESSTOKEN'
ansible-playbook -i inventory/hosts setup.yml --tags=setup-all,start --ask-become-pass

--->
Success. 
Hope this helps someone if you run into the same issue.
